neurots/preprocess/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Functions used for preprocessing."""
+from collections import defaultdict
+
+from neurots.generate.algorithms import tmdgrower
+from neurots.validator import validate_neuron_distribs
+from neurots.validator import validate_neuron_params
+
+_preprocess_functions = defaultdict(set)
+
+
+def register_preprocess(*growth_methods):
+    """Register a preprocess function."""
+
+    def inner(func):
+        for i in growth_methods:
+            _preprocess_functions[i].add(func)
+        return func
+
+    return inner
+
+
+@register_preprocess("trunk")
```

Comment:
I like this design, where input data are checked depending on the need. I'll get back to this once you integrate the rest of the comments.